bigquery/google/cloud/bigquery/job.py

```diff
@@ -742,7 +742,10 @@ def from_api_repr(cls, resource, client):
         dataset = Dataset(dest_config['datasetId'], client)
         destination = Table(dest_config['tableId'], dataset)
         sources = []
-        for source_config in config['sourceTables']:
+        source_configs = config.get('sourceTables')
+        if source_configs is None:
+            source_configs = [config['sourceTable']]
```

Comment:
Oh I think I see now. We default to [`sourceTables`](https://github.com/GoogleCloudPlatform/google-cloud-python/blob/a8fb99736bee614ad6d646853c003db15b941d50/bigquery/google/cloud/bigquery/job.py#L707) but if someone creates the job some other way then it *could* be `sourceTable`.

And yeah there shouldn't be any jobs coming back that are broken with neither set.